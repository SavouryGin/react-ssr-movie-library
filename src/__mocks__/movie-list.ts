export const movieList = [
  {
    title: 'Pulp Fiction',
    genres: ['Action', 'Adventure'],
    year: 2004,
    imagePath: '__mocks__/images/Bitmap-1.png',
  },
];
