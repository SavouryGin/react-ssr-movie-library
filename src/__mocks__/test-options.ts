import { SortingValues } from 'enums/params';
import { MultiSelectOption, SelectEntity } from 'types/controls';

export const mockSortOptions: SelectEntity[] = [
  {
    option: 'Test-1',
    value: 1,
  },
  {
    option: 'Test-2',
    value: 2,
  },
  {
    option: 'Test-3',
    value: 3,
  },
];

export const mockMultiSelectOptions: MultiSelectOption[] = [
  {
    label: 'Test-1',
    value: 1,
  },
  {
    label: 'Test-2',
    value: 2,
  },
  {
    label: 'Test-3',
    value: 3,
  },
];
