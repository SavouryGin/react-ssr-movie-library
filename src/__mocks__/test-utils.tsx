import React, { ReactElement, ReactNode } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { Reducer, configureStore } from '@reduxjs/toolkit';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { Field, Form } from 'react-final-form';

export const renderWithRedux = (ui: ReactElement, reducer: Reducer, preloadedState = {}, renderOptions?: RenderOptions): RenderResult => {
  const mockedStore = configureStore({ reducer, preloadedState });
  const Wrapper: React.FC = ({ children }: { children?: ReactNode }) => {
    return (
      <Provider store={mockedStore}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  };

  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export const renderInsideForm = (ui: ReactElement, renderOptions?: RenderOptions): RenderResult => {
  const Wrapper: React.FC = ({ children }: { children?: ReactNode }) => {
    return (
      <Form
        onSubmit={jest.fn()}
        render={() => {
          return <>{children}</>;
        }}
      />
    );
  };
  return render(ui, { wrapper: Wrapper, ...renderOptions });
};
