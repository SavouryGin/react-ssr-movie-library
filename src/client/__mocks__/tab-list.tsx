import React from 'react';
import { TabItem } from 'types/tabs';

export const testTabs: TabItem[] = [
  { tabContent: <>Tab content 1</>, tabId: 'tab-1', tabTitle: 'Title 1' },
  { tabContent: <>Tab content 2</>, tabId: 'tab-2', tabTitle: 'Title 2' },
  { tabContent: <>Tab content 3</>, tabId: 'tab-3', tabTitle: 'Title 3' },
  { tabContent: <>Tab content 4</>, tabId: 'tab-4', tabTitle: 'Title 4' },
];
