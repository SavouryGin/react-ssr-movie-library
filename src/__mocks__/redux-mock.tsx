import { combineReducers } from '@reduxjs/toolkit';
import { moviesInitialState, moviesSlice } from 'store/movies/slice';
import { movieList } from './movie-list';

export const mockedReducer = combineReducers({
  movies: moviesSlice.reducer,
});

export const mockedState = {
  movies: { ...moviesInitialState, list: movieList },
};
