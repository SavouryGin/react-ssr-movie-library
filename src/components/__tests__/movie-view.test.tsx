import MovieView from 'components/movie/view';
import React from 'react';
import { mockedReducer, mockedState } from '__mocks__/redux-mock';
import { renderWithRedux } from '__mocks__/test-utils';

const testProps = {
  movieId: 'test-id',
  onCloseView: jest.fn(),
  className: 'test-class',
};

describe('MovieView component:', () => {
  it('matches the snapshot', () => {
    const { asFragment } = renderWithRedux(<MovieView {...testProps} />, mockedReducer, mockedState);
    expect(asFragment()).toMatchSnapshot();
  });
});
