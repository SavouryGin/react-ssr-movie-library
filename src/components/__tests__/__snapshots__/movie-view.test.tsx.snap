// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieView component: matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="test-class"
  >
    <div
      class=""
    >
      <div
        style="width: 64px; height: 64px;"
      />
    </div>
    <button
      class="undefined"
      type="button"
    >
      <span
        class=""
      />
    </button>
    <div
      class=""
    >
      <img
        alt="Banner for \\"Not Found\\""
        role="img"
        src="assets/images/Fallback.jpg"
      />
    </div>
    <h2>
      Not Found 
      <span>
        0
      </span>
    </h2>
    <p />
    <p>
      2000 
      <span>
        Unknown
      </span>
    </p>
    <p />
  </div>
</DocumentFragment>
`;
